tsconfig.json

```diff
@@ -0,0 +1,46 @@
+{
+  /*
+    TypeScript is being used to do type checking across both JavaScript and TypeScript file.
+    In particular, this picks up some problems in the JSDoc in the JavaScript files, and validates the code
+    is consistent with the JSDoc.
+     
+    A complication is that TypeScript does not directly support different config for .ts/.js. To avoid drowning
+    in strict errors that are tedious to avoid in JavaScript the settings here are not strict. There is a
+    another config with strict settings for opt-in use.
+     
+    The settings here are used by VSCode, and TypeScript checks are enabled in included JavaScript files.
+    eslint supports custom config and uses the strict settings for TypeScript files.
+    To strictly check the published type definitions, there is a one script using the strict settings.
+  */
+  /* Visit https://aka.ms/tsconfig to read more about tsconfig configuration. */
+  "compilerOptions": {
+    "module": "commonjs",
+    "lib": [
+        "es6"
+    ],
```

Comment:
For interest, that comes from original PR: #646
(I'll make more comments later...

Reply:
On second thoughts, `lib` might be more appropriate because we aren't transpiring code here, but `target` works too because its value is used for `lib` by default.

Reply:
I'll look into that separately. Don't need to change that in this PR.

https://github.com/tsconfig/bases#node-16-tsconfigjson